### PR TITLE
Add date display format for 'Clock preferences'

### PR DESCRIPTION
**Pull request by LiuXingxiang-kylinsec**
Expected behaviour

    In China, we hope that like the Windows operating system, we can set the display format of time. Therefore, I added a long/short date display format for the Chinese environment.
    Long Date Display Format:YYYY年MM月DD日
    Short Date Display Format:YYYY-MM-DD

Actual behaviour
Steps to reproduce the behaviour
MATE general version

    git master master

Package version

    git master master

Linux Distribution

    Centos7.6

Link to downstream report of your Distribution

### CODE REVIEW

**Comment by LiuXingxiang-kylinsec** [comment]
I fix display error for longdate,and add a function to set default date format

**Comment by yetist** [comment]
I noticed that the clock applet supports custom format:

https://github.com/mate-desktop/mate-panel/blob/bbdaf1fd77e851bd0b63e4793e1f9309dbe12ebc/applets/clock/org.mate.panel.applet.clock.gschema.xml.in#L30

https://github.com/mate-desktop/mate-panel/blob/bbdaf1fd77e851bd0b63e4793e1f9309dbe12ebc/applets/clock/org.mate.panel.applet.clock.gschema.xml.in#L32

But there are no configuration widget on the UI, is it possible to add a third choice: 12-hours, 24-hours and custom format?

**Comment by lukefromdc** [comment]
I got this build error on rebuilding with the latest changes:
`
org.mate.panel.applet.clock.gschema.xml:57:1  Error on line 57 char 1: Element <key> not allowed inside <key>.  --strict was specified; exiting.`
build options were
`
./autogen.sh  --with-in-process-applets=wncklet,clock,fish,notification-area --prefix=/usr libdir=/usr/lib/x86_64-linux-gnu --sysconfdir=/etc --enable-shared=yes --enable-static=no --disable-maintainer-mode
`
Also I have no idea if file in Po are supposed to be changed as part of other PR's or if translaters do that separately, I know little of that part of the process but have never seen a change there as part of another PR that I can recall.

File: applets/clock/clock.c [hunk 1]
```diff
@@ -682,6 +773,64 @@ update_tooltip (ClockData * cd)
                 else
                         tip = _("Click to view month calendar");
         }
+        if (!cd->showdate) {
```
Comment (rbuj) [inline applets/clock/clock.c]: You don't need to show date and time on the tooltip, furthermore tip was set on the previous line.

File: applets/clock/clock.c [hunk 2]
```diff
@@ -146,6 +149,9 @@ struct _ClockData {
         ClockFormat  format;
         char        *custom_format;
         gboolean     showseconds;
+        gboolean     showlongdate;
+	gboolean     showshortdate;
+        gboolean     showdefaultdate;
```
Comment (rbuj) [inline applets/clock/clock.c]: You really only need a boolean variable, for instance showlongdate and set its default value.

**Comment by rbuj** [comment]
clock_format for short and long dates should be set in the else block, i.e. if cd->showdate is true.
https://github.com/mate-desktop/mate-panel/blob/8a4ae0116b1a75f6b77e60ef364288c38a60b35f/applets/clock/clock.c#L465-L489

**Comment by lukefromdc** [comment]
What is the status of this?

**Comment by raveit65** [comment]
@LiuXingxiang-kylinsec 
There are open remarks and a file conflict.

**Comment by raveit65** [comment]
@LiuXingxiang-kylinsec
Any news?

File: po/zh_CN.po [hunk 3]
```diff
@@ -313,6 +313,10 @@ msgstr "长日期格式：YYYY年MM月DD日"
 msgid "Show shortdate"
 msgstr "短日期格式：YYYY-MM-DD"
 
+#: ../applets/clock/clock.ui.h:37
```
Comment (raveit65) [inline po/zh_CN.po:316]: We have a file conflict with master branch at this point.
Better do not edit this file. Every pull from our translation sever (transifex) will override this file.
We will update mate-panel.pot file with your changes after this is merged.
After that the translators can update this translation string.

**Comment by lukefromdc** [comment]
Is this going to be fixed or should we close it? Lots of outstanbding PR's right now

**Pull request closed without merging**